examples/Mixtures/slot_based.py

```diff
@@ -0,0 +1,258 @@
+# mixture, i.e. the mixture slots are are considered permutation-invariant.
+
+# A complication with permutation invariance arises from the fact that we have not
+# only a label per slot, but also a numerical amount. If this amount is zero, then the
+# label of the slot becomes meaningless (i.e. which ingredient should be considered for the
+# slot), because adding zero of it does not change the mixture. In BayBE, we call
+# this a "dependency", i.e. the slot labels depend on the slot amounts and are only
+# relevant if the amount satisfies some condition (in this case "amount > 0").
+
+# The {class}`~baybe.constraints.discrete.DiscreteDependenciesConstraint` informs the
+# {class}`~baybe.constraints.discrete.DiscretePermutationInvarianceConstraint` about
+# these dependencies so that they are correctly included in the filtering process:
+
+perm_inv_constraint = DiscretePermutationInvarianceConstraint(
+    parameters=["Slot1_Label", "Slot2_Label", "Slot3_Label"],
+    dependencies=DiscreteDependenciesConstraint(
+        parameters=["Slot1_Amount", "Slot2_Amount", "Slot3_Amount"],
+        conditions=[
+            ThresholdCondition(threshold=0.0, operator=">"),
+            ThresholdCondition(threshold=0.0, operator=">"),
+            ThresholdCondition(threshold=0.0, operator=">"),
+        ],
+        affected_parameters=[["Slot1_Label"], ["Slot2_Label"], ["Slot3_Label"]],
+    ),
+)
+
+#### Substance Amounts
+
+# Interpreting the slot amounts as percentages, we need to ensure that their total is
+# always 100:
+
+sum_constraint = DiscreteSumConstraint(
+    parameters=["Slot1_Amount", "Slot2_Amount", "Slot3_Amount"],
+    condition=ThresholdCondition(threshold=100, operator="=", tolerance=SUM_TOLERANCE),
+)
+
+
+# We store all constraints in a single list:
+
+constraints = [perm_inv_constraint, sum_constraint, no_duplicates_constraint]
+
+
+### Search Space Creation
+
+# With all building blocks in place, we can now assemble our discrete space and inspect
+# its configurations:
+
+space = SubspaceDiscrete.from_product(parameters=parameters, constraints=constraints)
+print(space.exp_rep)
+
+### Verification of Constraints
+
+# Let us programmatically assert that all constraints are satisfied:
+
+amounts = space.exp_rep[["Slot1_Amount", "Slot2_Amount", "Slot3_Amount"]]
+labels = space.exp_rep[["Slot1_Label", "Slot2_Label", "Slot3_Label"]]
+
+
+# * All amounts sum to 100:
+
+n_wrong_sum = amounts.sum(axis=1).apply(lambda x: x - 100).abs().gt(SUM_TOLERANCE).sum()
+assert n_wrong_sum == 0
+print("Number of configurations whose amounts do not sum to 100: ", n_wrong_sum)
+
+
+# * There are no duplicate slot labels:
+
+n_duplicates = labels.nunique(axis=1).ne(3).sum()
+assert n_duplicates == 0
+print("Number of configurations with duplicate slot labels: ", n_duplicates)
+
+
+# * There are no permutation-invariant configurations:
+
+n_permute = labels.apply(frozenset, axis=1).to_frame().join(amounts).duplicated().sum()
```

Comment:
@Scienfitz: took this logic from your code, but could it be that this is actually wrong? Don't we need to build frozensets from tuples of (label, amount)?

Reply:
I think this is just in a [different place](https://github.com/emdgroup/baybe/blob/main/baybe/constraints/discrete.py#L244-L263) compared to [where you looked](https://github.com/emdgroup/baybe/blob/main/baybe/constraints/discrete.py#L315)

But for the test here it should be the pairs. check out the last commit

Reply:
I think I cannot follow what you try to explain here 🤔 All I can tell you: the check in the latest version is exactly what I was expecting to happen 👍🏼 But don't we agree that the check in the earlier version was something else?!?

Anyway: from my perspective, everything's fine now, so feel free to simply resolve. But if you think I missed something important in your explanation / the earlier version, let me know.